apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerParameter.cs

```diff
@@ -95,25 +129,23 @@ public override string ParameterName
         /// </summary>
         public SpannerDbType SpannerDbType
         {
-            get { return TypeCode.GetSpannerDbType(); }
-            set { TypeCode = value.GetSpannerTypeCode(); }
+            get;
+            set;
         }
 
         /// <inheritdoc />
         public override object Value
         {
-            get { return _value; }
+            get => _value;
             set
             {
-                if (TypeCode == TypeCode.Unspecified)
+                if (SpannerDbType.TypeCode == TypeCode.Unspecified && value != null)
```

Comment:
I wouldn't do this in the setter - it should be reasonable to set the properties in any order. I'd validate it on use.